src/jsonapi.js

```diff
@@ -140,25 +214,48 @@ export const updateResource = (resource) => {
 
           dispatch(apiUpdateFailed(err));
           reject(err);
-        });
+        })
+        .finally(() => dispatch(apiDidRequest(requestOptions)));
     });
   };
 };
 
-export const deleteResource = (resource) => {
+export const deleteResource = (resource, options = {}) => {
   return (dispatch, getState) => {
-    dispatch(apiWillDelete(resource));
+    const stateKey = options.stateKey || DEFAULT_STATE_KEY;
+    const { endpoint: { axiosConfig } } = getState()[stateKey];
+    const requestId = uniqueId();
+    const requestType = 'deleteResource';
+    const requestOptions = {
+      requestType,
+      requestId,
+      stateKey,
+      endpoint: `${resource.type}/${resource.id}`,
+      axiosConfig: {
+        method: 'DELETE',
+        ...axiosConfig,
+        ...(options.axiosConfig ? options.axiosConfig : {})
+      },
+      ...options
+    };
```

Comment:
unpacking `options` in the end, wouldn't that override `axiosConfig` as specified above, if present in `options`? i.e. rendering L237 useless. I don't see a test case for this, maybe that's relevant to add.

Reply:
True. Will change that.